src/containers/FactoryLoader.tsx

```diff
@@ -215,13 +229,42 @@ export class FactoryLoaderContainer extends React.PureComponent<Props, State> {
     this.setState({ currentStep: LoadFactorySteps.APPLYING_DEVFILE });
     await delay();
 
-    let workspace: che.Workspace | null = null;
-    try {
-      workspace = await this.props.createWorkspaceFromDevfile(devfile, undefined, undefined, attrs) as any;
-    } catch (e) {
-      this.showAlert(`Failed to create a workspace. ${e}`);
+    let workspace: che.Workspace | undefined | null = null;
+    if (this.state.createPolicy === 'peruser') {
+      workspace = this.props.allWorkspaces.find(workspace => {
+        return WorkspaceStatus[workspace.status] === WorkspaceStatus.RUNNING &&
+          workspace.attributes && workspace.attributes.stackName === attrs.stackName;
+      });
+      if (!workspace) {
+        workspace = this.props.allWorkspaces.find(workspace => {
+          return workspace.attributes && workspace.attributes.stackName === attrs.stackName;
+        });
+      }
+    } else if (this.state.createPolicy === 'peraccount') {
+      workspace = this.props.allWorkspaces.find(workspace => {
+        return WorkspaceStatus[workspace.status] === WorkspaceStatus.RUNNING &&
+          devfile && devfile.metadata && devfile.metadata.name === workspace.devfile.metadata.name;
```

Comment:
I don't think it's the right peraccount policy.

The only difference between perUser and perAccount,
perUser should loop over the current user workspaces
perAccount should loop over chosen account, could be useful for organizations, but at this point the only available account - is personal, so should not be any difference with preuser.

BTW I think we should be fine to remove peraccount for timebeing.

Reply:
I have removed 'peraccount' at all.